File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--adding-controls-to-a-document-at-run-time-in-a-vsto-add-in_3.vb
    Friend Sub ToggleRichTextControlOnDocument()

        Dim vstoDocument As Document = Globals.Factory.GetVstoObject(Me.Application.ActiveDocument)

        Dim name As String = "MyRichTextBoxControl"

        If Globals.Ribbons.MyRibbon.addRichTextCheckBox.Checked Then
            Dim selection = Me.Application.Selection
            If selection IsNot Nothing AndAlso selection.Range IsNot Nothing Then
                richTextControl = vstoDocument.Controls.AddRichTextContentControl( _
                        selection.Range, name)
            End If
        Else
            vstoDocument.Controls.Remove(name)
        End If
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--adding-controls-to-a-document-at-run-time-in-a-vsto-add-in_3.cs
        internal void ToggleRichTextControlOnDocument()
        {

            Document vstoDocument = Globals.Factory.GetVstoObject(this.Application.ActiveDocument);


            string name = "MyRichTextBoxControl";

            if (Globals.Ribbons.MyRibbon.addRichTextCheckBox.Checked)
            {
                Word.Selection selection = this.Application.Selection;
                if (selection != null && selection.Range != null)
                {
                    richTextControl = vstoDocument.Controls.AddRichTextContentControl(
                        selection.Range, name);
                }
            }
            else
            {
                vstoDocument.Controls.Remove(name);
            }
        }